kong/router/atc.lua

```diff
@@ -382,12 +383,17 @@ function _M:matching(params)
 
   local request_prefix = matched_route.strip_path and matched_path or nil
 
+  local uri_captures = nil
+  if matched_path and captures and pl_tablex.size(captures) > 1 then
```

Comment:
should we use `not table.isempty()` to replace `pl_tablex.size`?

Reply:
> This place requires the size of `captures` to be > 1, which means capturing at least one element from the URI. So guess we cannot use table.isempty here, is that right? @StarlightIbuki

I'm unsure if `captures[0]` is used to return the whole match here, but I suppose it's not. The original condition was to check if there's at least 1 match, thus if I'm right, we need to use `is_emtpy` instead.

@chronolaw any idea?

Reply:
I think that we should use `table.isempty`, see: https://github.com/Kong/atc-router/blob/main/lib/resty/router/context.lua#L76

Reply:
Could we remove the condition `matched_path`? If no matching the captures must be `nil`

Reply:
This line could be :
```
local uri_captures = (captures and not tb_isempty(captures)) and captures or nil
```

Reply:
> I'm unsure if captures[0] is used to return the whole match here, but I suppose it's not.

Yes, captures[0] will return the whole match. And change to is_empty might break the old behavior, like this test case: https://github.com/Kong/kong/blob/master/spec/01-unit/08-router_spec.lua#L1084

The current code will return a nil `uri_capatured`, but changed to `table.isempty` will become not nil(uri_capature[0]). Not sure if it's expected, if yes, then I can change the corresponding test cases as well:

```
Failure → spec/01-unit/08-router_spec.lua @ 1055
Router (flavor = expressions) select() [uri regex] matches a [uri regex] even if a [prefix uri] got a match
spec/01-unit/08-router_spec.lua:1084: Expected objects to be the same.
Passed in:
(table: 0x01367e2758) {
  [0] = '/route/persons/123/profile' }
Expected:
(nil)

stack traceback:
        spec/01-unit/08-router_spec.lua:1084: in function <spec/01-unit/08-router_spec.lua:1055>
```

Reply:
cc @StarlightIbuki @chronolaw

Reply:
@git-hulk Please revert my commit

Reply:
OK, I didn't apply those commits yet.

Reply:
I got. `captures[0]` should not be treated as a capture, right?

Reply:
Yes, that's why I check if the table size > 1 instead of >= 1

Reply:
Then we should use `table.nkeys`, it is better than pl.tablex

Reply:
@chronolaw Thanks, will change to `table.nkeys`

Reply:
I think that `>= 2` is more clear for reviewer, it will reminder the people there are two capture results.

Reply:
We should also add a comment for this condition, it is very easy to confuse the reviewer.